src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -581,14 +597,15 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
         }
 
         /// <summary>
-        /// Symbol for an out variable local.
+        /// Symbol for an out variable local that might require type inference during overload resolution, i.e.
+        /// its type is syntactically 'var'.
         /// </summary>
-        private class OutLocalSymbol : SourceLocalSymbol
+        private class PossiblyImplicitlyTypedOutVarLocalSymbol : SourceLocalSymbol
```

Comment:
No, it doesn't violate a policy, just good design sense. We don't call it `PossiblyImplicitlyTypedLocalSymbolWithInitializer`, and for the same reason we should retain the name you deleted here.